src/main/java/de/tum/in/www1/artemis/service/exam/ExamUserService.java

```diff
@@ -106,26 +109,31 @@ public ExamUsersNotFoundDTO saveImages(long examId, MultipartFile file) {
         List<String> notFoundExamUsersRegistrationNumbers = new ArrayList<>();
         List<ExamUserWithImageDTO> examUserWithImageDTOs = parsePDF(file);
 
-        examUserWithImageDTOs.forEach(examUserWithImageDTO -> {
+        for (var examUserWithImageDTO : examUserWithImageDTOs) {
             Optional<User> user = userRepository.findUserWithGroupsAndAuthoritiesByRegistrationNumber(examUserWithImageDTO.studentRegistrationNumber());
-            if (user.isPresent()) {
-                Optional<ExamUser> examUserOptional = examUserRepository.findByExamIdAndUserId(examId, user.get().getId());
-                if (examUserOptional.isPresent()) {
-                    ExamUser examUser = examUserOptional.get();
-                    MultipartFile studentImageFile = fileService.convertByteArrayToMultipart(examUserWithImageDTO.studentRegistrationNumber() + "_student_image", ".png",
-                            examUserWithImageDTO.image().imageInBytes());
-                    String responsePath = fileService.handleSaveFile(studentImageFile, false, false).toString();
-                    examUser.setStudentImagePath(responsePath);
-                    examUserRepository.save(examUser);
-                }
-                else {
-                    notFoundExamUsersRegistrationNumbers.add(examUserWithImageDTO.studentRegistrationNumber());
-                }
+            if (user.isEmpty()) {
+                notFoundExamUsersRegistrationNumbers.add(examUserWithImageDTO.studentRegistrationNumber());
+                continue;
             }
-            else {
+            Optional<ExamUser> examUserOptional = examUserRepository.findByExamIdAndUserId(examId, user.get().getId());
+            if (examUserOptional.isEmpty()) {
                 notFoundExamUsersRegistrationNumbers.add(examUserWithImageDTO.studentRegistrationNumber());
+                continue;
             }
-        });
+
+            ExamUser examUser = examUserOptional.get();
+            String oldPathString = examUser.getStudentImagePath();
+            Path pathToSave = FilePathService.getStudentImageFilePath().resolve(examUserWithImageDTO.studentRegistrationNumber() + "_student_image.png");
```

Comment:
So the filename is just the registration number plus the constant string? 

Could this lead to issues if the same user is registered for two exams, and then unregisters for one of the two? Since deleting the image then deletes the image for both exams.

Reply:
Good catch. I misinterpreted the original code here